indigo_app/views.py

```diff
@@ -261,6 +269,142 @@ def post(self, request, frbr_uri, version_id):
         return redirect(url)
 
 
+class BatchAddWorkView(AbstractAuthedIndigoView, TemplateView):
+    template_name = 'work/new_batch.html'
+    # permissions
+    permission_required = ('indigo_api.add_work',)
+
+    def get_context_data(self, **kwargs):
+        context = super(BatchAddWorkView, self).get_context_data(**kwargs)
+
+        def get_works():
+
+            table = get_table()
+            works = []
+
+            for idx, row in enumerate(table):
+                # TODO: match header row contents rather than using column numbers
+                row_number = idx+1
+
+                info = {
+                    'row': row_number,
+                }
+
+                frbr_uri = get_frbr_uri(row)
+
+                # TODO: fix get_uri to return uri as false if validation error?
+
+                if frbr_uri:
+
+                    try:
+                        work = Work.objects.get(frbr_uri=frbr_uri)
+                        info['work'] = work
+                        info['status'] = 2
+
+                    # TODO one day: also mark another work as duplicate if user is trying to import two of the same (currently only the second one will be '2')
+
+                    except Work.DoesNotExist:
+
+                        work = Work()
+
+                        work.frbr_uri = frbr_uri
+                        work.title = row[8]
+                        work.country = row[9]
+                        work.publication_name = row[10]
+                        work.publication_number = row[11]
+                        work.publication_date = row[12]
```

Comment:
These will need to be parsed from strings into dates. Python has some support for this. Let's see what format we get from google sheets, first.